### DISCUSSION TITLE
Adding only and except methods

### DISCUSSION THREAD

**Discussion by agentphoenix** [spatie/laravel-data#5] [Ideas]
Really enjoying this library! I'd like to use it to replace my DTOs, but I'm finding that I'm missing the `only` and `except` methods that your DTO package has.

```php
// Currently doing this...
Arr::except($data->all(), 'field');

// Would love to be able to do this instead...
$data->except('field')->toArray();
```

I see that there are `include` and `exclude` methods, but it's not clear if that's meant for the same thing in this package.

Would you guys be open to a PR adding the `only` and `except` methods into the package?

**Comment by freekmurze** [comment]
I think it makes sense to add such methods.

What do you think @rubenvanassche ?

**Comment by IsraelOrtuno** [comment]
I was kind of looking for the same, I guess a `UserData` may have a `password` property but it won't ever be returned by any resource. Is there a way to do this as of now?

**Reply by IsraelOrtuno** [reply]
Or may even make sense to have a `UserResource` class which does the exclusion for these particular use cases?

**Comment by rubenvanassche** [comment]
@agentphoenix That seems like a great idea! Though it it will take some work to implement, the include and exclude methods allow you to include/exclude lazy properties. You need to also implement the `only` / `except` methods which need to decide in advance which properties should be transformed and which not. 

I think `only`/`except` should have precedence on the lazy properties. So if someones hides a property with `except` and then includes that same property then it shouldn't be in the payload.

It would be really cool if you'd use the `PartialsParser` to make nested things like this possible:

```php
ArtistData::from($artist)->only('name', 'songs.title')->toArray();
```

I'd love to see such a PR! Good luck!

**Reply by agentphoenix** [reply]
Awesome! Thanks for the insight @rubenvanassche. I'll definitely jump in and take a shot at adding the methods.
